### ISSUE TITLE
Custom data grid columns not configurable

### ISSUE THREAD

**Issue by vcondratiev** [formio/formio.js#5013]
Hello, I'm trying to create a custom data grid component by extending the existing data grid.
What I'm trying to do is to add a new column with some buttons/actions for each row and also load some data from an api.

I created a new component and extended the existing one.
My problem right now is that after I register my component and try to use it in my builder I can't configure the columns.

`import { Components } from "formiojs";

const DataGridComponent = Components.components.datagrid;
import editForm from 'formiojs/components/datagrid/DataGrid.form';

export default class CustomDataGrid extends DataGridComponent {
  constructor(args, options, data) {
    super(args, options, data);
  }

  init() {
    super.init();
  }

  static editForm = editForm;

  static schema(...extend) {
    let newSchema = { extend, ...{
      type: 'customdatagrid',
      key: 'customdatagrid',
      label: 'Custom Data Grid',
      input: true,
      tree: true,
      components: []
    } }
    return DataGridComponent.schema(newSchema);
  }

  static get builderInfo() {
    return {
      title: 'Custom Data Grid',
      group: 'data',
      icon: 'fa fa-table',
      weight: 70,
      documentation: '/userguide/forms/data-components#data-grid',
      schema: CustomDataGrid.schema()
    };
  }

  render() {
    return super.render(this.renderTemplate('customdatagrid', {}));
  }

  attach(element) {
    return super.attach(element);
  }

  detach() {
    return super.detach();
  }

  destroy() {
    return super.destroy();
  }

  getValue() {
    return super.getValue();
  }

  setValue(value, flags = {}) {
    return super.setValue(value, flags);
  }
}

CustomDataGrid.editForm = editForm;

Components.addComponent('customdatagrid', CustomDataGrid);
`
Basically I can't drag any components in my data grid and to configure the grid. I can change the settings, it is the same as a normal data grid.
Am I missing something ? Why can't I configure my custom data grid in form builder?

Thank you

**Comment by vcondratiev** [comment]
Found exact issue: https://github.com/formio/formio.js/issues/2936
Will close this one.
Thank you
